tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -586,6 +586,105 @@ public function test_get_normalized_script_args() {
 		$this->assertSame( $expected_args, $wp_scripts->get_data( 'footer-old', 'script_args' ) );
 	}
 
+	/**
+	 * Test script concatenation with deferred main script.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_concatenate_with_defer_strategy() {
+		global $wp_scripts, $concatenate_scripts;
+
+		$old_value = $concatenate_scripts;
+		$concatenate_scripts = true;
+
+		$wp_scripts->do_concat    = true;
+		$wp_scripts->default_dirs = array( '/directory/' );
+
+		wp_register_script( 'one-concat-dep', '/directory/script.js' );
+		wp_register_script( 'two-concat-dep', '/directory/script.js' );
+		wp_register_script( 'three-concat-dep', '/directory/script.js' );
+		wp_enqueue_script( 'main-defer-script', '/main-script.js', array( 'one-concat-dep', 'two-concat-dep', 'three-concat-dep' ), null, array( 'strategy' => 'defer' ) );
+
+		wp_print_scripts();
+		$print_scripts = get_echo( '_print_scripts' );
+
+		// reset global before asserting.
+		$concatenate_scripts = $old_value;
+
+		$ver       = get_bloginfo( 'version' );
+		$expected  = "<script type='text/javascript' src='/wp-admin/load-scripts.php?c=0&amp;load%5Bchunk_0%5D=one-concat-dep,two-concat-dep,three-concat-dep&amp;ver={$ver}'></script>\n";
+		$expected .= "<script type='text/javascript' src='/main-script.js' id='main-defer-script-js' defer></script>\n";
+
+		$this->assertSame( $expected, $print_scripts );
+	}
+
+	/**
+	 * Test script concatenation with `async` main script.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_concatenate_with_async_strategy() {
+		global $wp_scripts, $concatenate_scripts;
+
+		$old_value = $concatenate_scripts;
```

Comment:
Since we are modifying the global `$wp_scripts` variable directly, I wonder if we even need to modify the global `$concatenate_scripts` variable? We may also need to reset the global `$wp_scripts` variable before running or assertions.

Reply:
@joemcgill This is the case with almost every test case. Infact, most of the test case contains handles from previous test cases. I followed this rule -> 
as long as this doesn't affect other test cases, not resetting global should be fine.

This function was created by replicating an old concatenation test function, setting of the directory was also taken from it.

cc: @10upsimon.

Reply:
Thanks for the context Karthik. I’ll make a note to review our use of that global in all the tests. Not something we need to address here.